src/primitive-field/primitive-field.component.ts

```diff
@@ -72,15 +71,16 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
 
   commitValueChange() {
     // Validation
-    if (this.schema['type'] === 'string' && this.schema['enum'] === undefined) {
-      try {
-        this.schemaValidationService.validateStringValue(this.value.toString(), this.schema);
-      } catch (error) {
-        console.error(error);
-      }
+    let errors = [];
+    let {isValid, error} = this.schemaValidationService.validateValue(this.value, this.schema);
+    if (isValid) {
+      this.jsonStoreService.setIn(this.path, this.value);
+    } else {
+      errors = [{
+        message: error
+      }];
     }
-    // TODO: should we make the change even if it is not validated
-    this.jsonStoreService.setIn(this.path, this.value);
+    this.errors = errors;
```

Comment:
well, we need to have something otherwise we loose one of them everything the other is updated.